app/helpers/metrics_helper.rb

```diff
@@ -0,0 +1,101 @@
+module MetricsHelper
+  def parse_agent_data(agents)
+    agents.map do |a|
+      {
+        x: a.label,
+        y: a.count
+      }
+    end
+  end
+
+  def sum_rollups(rollups)
+    rollups.map { |r| r[:count] }.reduce(:+)
+  end
+
+  def interval_options
+    Rollups::HourlyDownload::INTERVALS.map { |i| [I18n.t(".helpers.label.metrics.interval.#{i.downcase}"), i] }
+  end
+
+  def date_range_options(episode = nil)
+    if episode
+      episode_date_presets(episode)
+    else
+      podcast_date_presets
+    end
+  end
+
+  def podcast_date_presets
+    Rollups::HourlyDownload::PODCAST_DATE_PRESETS.map do |opt|
+      count, interval = opt.to_s.split("_")
+      date_end = Date.utc_today
+      date_start = if count.to_i > 0
+        count.to_i.send(interval).ago.utc_date
+      else
+        Date.utc_today.send(:"beginning_of_#{interval}")
+      end
+
+      [I18n.t(".helpers.label.metrics.date_presets.#{opt}"), date_preset(date_start, date_end)]
+    end
+  end
+
+  def episode_date_presets(episode)
+    Rollups::HourlyDownload::EPISODE_DATE_PRESETS.map do |opt|
+      count, interval, type = opt.to_s.split("_")
+
+      date_start = if type == "last"
+        count.to_i.send(interval).ago.utc_date
+      elsif count == "date"
+        Date.utc_today.send(:"beginning_of_#{interval}")
+      else
+        episode.utc_publish_date
+      end
+
+      date_end = if type == "drop"
+        date_start + count.to_i.send(interval)
+      else
+        Date.utc_today
+      end
+
+      [I18n.t(".helpers.label.metrics.date_presets.#{opt}"), date_preset(date_start, date_end, episode)]
+    end
+  end
+
+  def guard_date_start(date_start, episode = nil)
+    if episode && date_start < episode.utc_publish_date
+      episode.utc_publish_date
+    else
+      date_start
+    end
+  end
+
+  def guard_date_end(date_end)
+    if date_end > Date.utc_today
+      Date.utc_today
+    else
+      date_end
+    end
+  end
+
+  def date_preset(date_start, date_end, episode = nil)
+    [
+      guard_date_start(date_start, episode),
+      guard_date_end(date_end)
+    ].to_json
+  end
+
+  def date_filter_min(episode = nil)
```

Comment:
UX difference between podcast metrics and episode metrics. the minimum date in podcast metrics is set to 18 months back (arbitrary). however, in episode metrics, there is a desire to be able to see `All-time Downloads`, meaning from publish date to today. this means that the minimum boundary is different per episode.

potential consideration here is for older episodes; the minimum will constantly grow further into the past. for example: a user would be able to see 3 years of downloads data in the episode metrics view, and would eventually apply to the `agents` and `geos` cards when those are developed. is this something we want in Feeder? or should we consider a tighter scope of what the user has access to?

Reply:
potential ideas for solution:
- remove or limit the episode metrics date presets
- consider re-adding `All-time Downloads` tab that doesn't respond to the date selector
- consider adding `Drop day` tab that also doesn't respond to the date selector (similar to podcast metrics)